packages/gestalt/src/SearchField.js

```diff
@@ -78,46 +61,35 @@ export default class SearchField extends React.Component<Props, State> {
     }
   };
 
-  handleBlur = (event: SyntheticEvent<HTMLInputElement>) => {
-    const { onBlur } = this.props;
-    this.setState({ focused: false });
+  const handleBlur = (event: SyntheticEvent<HTMLInputElement>) => {
+    setFocused(false);
 
     if (onBlur) {
       onBlur({ event });
     }
   };
 
-  render() {
-    const {
-      accessibilityLabel,
-      autoComplete,
-      id,
-      placeholder,
-      size = 'md',
-      value,
-    } = this.props;
-
-    const { focused, hovered } = this.state;
-
-    // This mirrors the built in browser behavior. If there's a value, show the
-    // clear button if you're hovering or if you've focused on the field
-    const showClear = (focused || hovered) && value && value.length > 0;
-
-    const className = classnames(
-      styles.input,
-      size === 'md' ? layout.medium : layout.large
-    );
-
-    return (
-      <Box
-        display="flex"
-        position="relative"
-        alignItems="center"
-        onMouseEnter={this.handleMouseEnter}
-        onMouseLeave={this.handleMouseLeave}
-        onFocus={this.handleFocus}
-        onBlur={this.handleBlur}
-      >
+  const isNotEmpty = value && value.length > 0;
```

Comment:
@nachofassini `isNot...` can be confusing when reading through the code. Can we rename `isNotEmpty` to `hasValue`?

Reply:
Fixed!